src/TypeSpec.Extension/Emitter.Csharp/src/lib/operation.ts

```diff
@@ -400,6 +404,13 @@ export function loadOperation(
                 // for now, let assume we don't allow return type
                 StatusCodes: op.verb === "delete" ? [204] : [200],
                 BodyType: bodyType,
+                ResultPath:
+                    metadata.logicalPath ??
+                    // https://github.com/Azure/typespec-azure/issues/3725
+                    (metadata.envelopeResult != metadata.logicalResult &&
+                    op.verb === "post"
+                        ? "result" // actually `result` is the only allowed path for now
```

Comment:
Can this be more than just one property?

Reply:
@AlexanderSher Currently it should be only one property. I used to have a snippet of codes which will parse the `logicalPath` and convert into a property expression like `A.B.C`. After discussing with people more familiar with the current implementation of LRO in typespec, I abandoned the codes to keep it simple. We can add the logic back if needed, it's not complex.